### PR TITLE
Test set for automatic time tracking of follow-ups

### PR DESCRIPTION
**Pull request by samsplunks**
Creating a set of tests for the time tracking features.

### CODE REVIEW

**Comment by samsplunks** [comment]
Hi guys, found some time for testing the time_spent calculation system.

Helped me found a bug and fixed it inside this same PR.

**Comment by uhurusurfa** [comment]
> Hi guys, found some time for testing the time_spent calculation system.
> 
> Helped me found a bug and fixed it inside this same PR.

.. and thats why we love unit tests. 🥳

**Review by uhurusurfa — APPROVED** [review]
Thanks for the new fearture and the unit test

**Review by Benbb96 — APPROVED** [review]

File: helpdesk/tests/test_time_spent_auto.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                date=fup_time_p,
+                title="Testing followup",
+                comment="Testing followup time spent",
+                public=True,
+                user=self.user,
+                new_status=1,
+                message_id=uuid.uuid4().hex,
+                time_spent=None
+            )
+            followup1.save()
```
[the remark this answers is not in the document]
Comment (samsplunks) [reply helpdesk/tests/test_time_spent_auto.py:75]: Hi @Benbb96 you are totally right.
I blindly copied the code from another test and the create method is enough.

File: helpdesk/lib.py [hunk 2]
```diff
@@ -230,8 +236,6 @@ def daily_time_spent_calculation(earliest, latest, open_hours):
     day_delta = latest - earliest
 
     # returns up to 86399 seconds, add one second if full day
```
[the remark this answers is not in the document]
Comment (samsplunks) [reply helpdesk/lib.py:238]: Nope.

File: helpdesk/tests/test_time_spent_auto.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+            slug='q1',
+            allow_public_submission=True,
+            dedicated_time=timedelta(minutes=60)
+        )
+
+        self.ticket_data = dict(queue=self.queue_public,
+                                title='test ticket',
+                                description='test ticket description')
+
+        self.client = Client()
```
[the remark this answers is not in the document]
Comment (samsplunks) [reply helpdesk/tests/test_time_spent_auto.py:28]: Correct.

I will clean the code with your great remarks soon.

**Pull request merged**